Katana/Store.swift

```diff
@@ -433,16 +468,15 @@ private extension Store {
    */
   private func initializeInternalState(using stateInizializer: StateInitializer<S>) {
     self.state = stateInizializer()
-    // invoke listeners (always in main queue)
-    DispatchQueue.main.async {
-      self.listeners.values.forEach { $0() }
-    }
     self.initializedInterceptors = Store.initializedInterceptors(self.interceptors, sideEffectContext: self.sideEffectContext)
     
     // and here we are finally able to start the queues
     self.isReady = true
     self.sideEffectQueue.resume()
     self.stateUpdaterQueue.resume()
+    
+    // invoke listeners (we already are in the main queue)
```

Comment:
```suggestion
    // invoke listeners
```

mmh im not sure i'd mention that we are in the main queue, since that depends on what you assign to the AsyncProvider (only the default is actually the DispatchQueue.main)